docs/design/proposals/labels.md

```diff
@@ -0,0 +1,83 @@
+
+#### Overview
+
+The proposal is to add the following labels:
+-   A label on an instance to indicate which RGD is reconciling it.
+-   A label on resources created by an instance reconciliation to indicate which RGD they come from.
+-   Labels on resources created by an instance reconciliation to indicate which instance created them.
+
+This aligns with the common Kubernetes practice of using labels for discovery and organization of related resources, as seen in tools like Helm and Kustomize.
+
+#### Design details
+
+We will introduce the following new labels, which will be defined in `@pkg/metadata/labels.go`:
+
+1.  **Label for an instance being reconciled:**
+    -   **Proposed Label:** `kro.run/reconciled-by`
+    -   **Value:** The name of the ResourceGraphDefinition (e.g., `my-rgd-name`).
+    -   **Purpose:** To indicate that an instance is being actively reconciled against a specific RGD. This label will be applied to the instance itself.
+    -   **Alternative Names:** `kro.run/reconciling-rgd`, `kro.run/template`.
+
+2.  **Label for resources created during instance reconciliation:**
+    -   **Proposed Label:** `kro.run/created-by`
+    -   **Value:** The name of the ResourceGraphDefinition (e.g., `my-rgd-name`).
+    -   **Purpose:** To identify which RGD was used as a template to create a resource during the reconciliation of an instance. This label will be applied to all resources created by the instance controller.
+    -   **Alternative Names:** `kro.run/managed-by-rgd`, `kro.run/owner-rgd`, `kro.run/template-rgd`. The `managed-by` label is a common pattern in the Kubernetes ecosystem.
+
+3.  **Labels for resources to link back to the instance:**
+    -   **Proposed Labels:** 
+        - `kro.run/managed-by-instance-group`: The API group of the instance (e.g., `mygroup.example.com`).
+        - `kro.run/managed-by-instance-kind`: The kind of the instance (e.g., `MyKind`).
+        - `kro.run/managed-by-instance-namespace`: The namespace of the instance (e.g., `default`).
+        - `kro.run/managed-by-instance-name`: The name of the instance (e.g., `my-instance`).
+        - The `managed-by` label is a common pattern in the Kubernetes ecosystem.
+    -   **Purpose:** To provide a direct and queryable link from a created resource back to the instance that caused its creation. This set of labels uniquely identifies the instance.
+    -   **Alternative Name prefix:** `kro.run/owner-`, `kro.run/created-by-`, etc.
+
+## Other solutions considered
+
+We could continue using the existing labels, but their purpose is not as explicit for relationship tracking, which can lead to confusion for users and client tools. We could also use annotations, but labels are better suited for this purpose as they are queryable via the Kubernetes API, which is a key requirement for observability and tooling.
```

Comment:
nit: The existing labels should maybe listed here but honestly its fine for this proposal